### ISSUE TITLE
Line wrapping behavior on ASA (still happening)

### ISSUE THREAD

**Issue by ktbyers** [ktbyers/netmiko#281]
###### 

object-group network PYCLASS
twb-dc-fw1(config-network-object-group)# no network-object object pylab7a.twb-$
twb-dc-fw1(config-network-object-group)# no object network pylab7a.twb-tech.com
twb-dc-fw1(config)# end
twb-dc-fw1# 
###### 

Netmiko 1.0.0

**Comment by lykinsbd** [comment]
Just so we're clear, is the issue that you're encountering the `$` line truncation?

In looking at [cisco_asa_ssh.py, line 20](https://github.com/ktbyers/netmiko/blob/develop/netmiko/cisco/cisco_asa_ssh.py#L20) it appears that we're trying to set the terminal width in `Enable` mode.

However, because the ASA is strange you can only control the `terminal pager` length in `Enable` mode.  To control the `terminal width` is actually a configuration change that requires you to enter `config` mode.  Example below:

```
FW-LAB-BETA# terminal ?

  interactive  enable '?' as help
  monitor      Syslog monitor
  no           Turn off syslogging to this terminal
  pager        Control page length for pagination. The page length set here is not saved to configuration.
FW-LAB-BETA# conf t
FW-LAB-BETA(config)# terminal ?

configure mode commands/options:
  width  Set terminal width, default is 80 columns

exec mode commands/options:
  interactive  enable '?' as help
  monitor      Syslog monitor
  no           Turn off syslogging to this terminal
  pager        Control page length for pagination. The page length set here is not saved to configuration.
FW-LAB-BETA(config)# end
FW-LAB-BETA#
```

**Comment by ktbyers** [comment]
@lykinsbd Yes, that looks right (i.e. that is an error) and it needs done in config mode.

That is annoying that the ASA does that...

**Comment by lykinsbd** [comment]
Should I submit a PR to do one of the following:

1. Remove the attempt to set Terminal Width on the ASA
2. Change the Terminal Width method on the ASA class to execute via dropping into config mode

On the one hand, if there is output cut off on-screen, it may not serve people's purpose.

But on the other I don't know if it would be good to attempt config modifications without the user explicitly requesting it.

**Comment by ktbyers** [comment]
Yes, I think we should change the terminal width and document it...but agreed don't really like it.

I would say when we disconnect we reset it back to the default width.

So something like:
1. Check if we can get into config mode. If not, do nothing here.
2. If we can, set the terminal width. Save the previous terminal width in an attribute (default terminal width). Don't change it if is already right (i.e. don't change it if already the value it should be).

I want to experiment a bit and make sure that having a longer terminal width doesn't have adverse behavior when you use the CLI.

If it does cause negative side effects for CLI users, then we should set it back to the default value of 80 when we disconnect.

**Comment by lykinsbd** [comment]
I can confirm (that in my experience) it doesn't cause any issues to have terminal width set to 511.  We run a fleet of 18,000 ASAs at work and haven't seen any ill effects from it.

However, I agree with you your logic above.  Especially on saving/setting it back to previous value.  "Leave No Trace" on the config, so to speak.

**Comment by Rosiak** [comment]
Hi there,
Anything I can help with to improve progress of this issue?(Code, testing etc)

**Comment by ktbyers** [comment]
Should be fixed in PR #906
